aya/src/programs/mod.rs

```diff
@@ -955,7 +955,7 @@ impl ProgramInfo {
                 call: "bpf_prog_get_fd_by_id",
                 io_error,
             })?;
-        Ok(fd as RawFd)
+        Ok(fd.into_raw_fd())
```

Comment:
Here is another place that this PR probably broke things. The `fd` returned here is already closed because the `OwnedFd` is dropped so anyone calling for this `fd` is getting a bad file descriptor. Additionally, the comment of this method no longer makes sense:

```
/// The returned fd must be closed when no longer needed.
```

I think this would be a quick fix though, we just gotta make this function return an `OwnedFd` instead and remove the comment since `OwnedFd`s are closed on drop.

Reply:
Derp nvm, I thought we were calling for `as_raw_fd`. Ignore